incentivization.md

```diff
@@ -0,0 +1,206 @@
+- Cost calculation - see https://github.com/waku-org/research/issues/35
+- Price advertisement - see https://github.com/waku-org/research/issues/51
+- Price negotiation - see https://github.com/waku-org/research/issues/52
+
+## Payment
+
+If the client agrees to the price, it sends a _proof of payment_ to the server.
+For the MVP, each request is paid for with a separate blockchain transaction.
+The transaction hash (`txid`) acts as a proof of payment.
+
+Note that client gives proof of payment before it receives the response.
+Other options could be:
+1. the client pays after the fact;
+2. the client pays partly upfront and partly after the fact;
+3. an escrow (a centralized trusted third party, or a semi-trusted entity like a smart contract) ensures atomicity .
+
+Our design considerations are:
+- the MVP protocol should be simple;
+- servers are more "permanent" entities and are more likely to have long-lived identities;
+- it is more important to protect the clients's privacy than the server's privacy.
+
+In light of these criteria, we suggest that the client pays first: this is simpler than splitting the payment, more secure than trusting a third party, and (arguably) more privacy-preserving for the client than the alternative where the client pays after the fact (that would encourage servers to deanonymize clients to prevent fraud).
+
+### Future work
+
+- Add more payment methods - see https://github.com/waku-org/research/issues/58
+- Design a subscription model with service credentials - see https://github.com/waku-org/research/issues/59
+- Add privacy to service credentials - see https://github.com/waku-org/research/issues/60
+
+## Reputation
+
+We use reputation to discourage the server from taking the payment and not responding.
+The client keeps track of the server's reputation:
+- all servers start with zero reputation;
+- if the server honors the request, it gets +1;
+- if the server does not respond before the payment, it gets -1;
+- if the server does not respond after the payment and before a timeout, the client will never query it again.
+
+Potential issues:
+- An attacker can establish new server identities and continue running away with clients' money. Countermeasures:
+	- a client only queries "trusted" servers (which however leads to centralization);
+	- when querying a new server, a client first sends a small (i.e. cheap) request as a test.
+- The ban mechanism can theoretically be abused. For instance, a competitor may attack the victim server and cause the clients who were awaiting the response to ban that server. Countermeasure: prevent DoS-attacks.
+- Servers may also farm reputation by running clients and querying their own server.
+
+### Future work
+
+Design a more comprehensive reputation system:
+- local reputation - see https://github.com/waku-org/research/issues/48
+- global reputation - see https://github.com/waku-org/research/issues/49
```

Comment:
this seems quite difficult since it involves consensus on reputation, which adds a new dimension. Can't actually think of any web3 protocol with such feature.

Reply:
Agree, but that's the question for future research. We may conclude as a result of this research that we don't need / don't want / can't afford to implement a global reputation, which is indeed likely.